### PR TITLE
Update with okta-react v4 OKTA-340163

### PR DESCRIPTION
**Pull request by shuowu-okta**

### CODE REVIEW

File: custom-login/package.json [hunk 1]
```diff
@@ -3,7 +3,8 @@
   "version": "0.2.0",
   "private": true,
   "dependencies": {
-    "@okta/okta-react": "^3.0.8",
+    "@okta/okta-auth-js": "^4.1.0",
+    "@okta/okta-react": "file:.yalc/@okta/okta-react",
```
Comment (shuowu-okta) [inline custom-login/package.json:7]: update after 4.0 release

File: okta-hosted-login/package.json [hunk 2]
```diff
@@ -3,7 +3,8 @@
   "version": "0.2.0",
   "private": true,
   "dependencies": {
-    "@okta/okta-react": "^3.0.7",
+    "@okta/okta-auth-js": "^4.1.0",
+    "@okta/okta-react": "file:.yalc/@okta/okta-react",
```
Comment (shuowu-okta) [inline okta-hosted-login/package.json:7]: update after 4.0 release

File: custom-login/package.json (hunk 1 above)
Comment (swiftone) [reply custom-login/package.json:7]: Can we keep this PR as Draft until this is updated?

File: custom-login/src/Login.jsx [hunk 3]
```diff
@@ -46,20 +46,15 @@ const Login = () => {
     widget.renderEl(
```
Comment (swiftone) [inline custom-login/src/Login.jsx:46]: Should we be showing a recommended way other than renderEl?

File: custom-login/src/Navbar.jsx [hunk 4]
```diff
@@ -12,32 +12,38 @@
...
-          {authState.isAuthenticated && <Menu.Item id="profile-button" as="a" href="/profile">Profile</Menu.Item>}
-          {authState.isAuthenticated && <Menu.Item id="logout-button" as="a" onClick={logout}>Logout</Menu.Item>}
-          {!authState.isPending && !authState.isAuthenticated && <Menu.Item as="a" onClick={login}>Login</Menu.Item>}
+          {authState.isAuthenticated && (
+            <Menu.Item id="profile-button">
+              <Link to="/profile">Profile</Link>
+            </Menu.Item>
+          )}
+          {authState.isAuthenticated && <Menu.Item id="logout-button" onClick={logout}>Logout</Menu.Item>}
+          {!authState.isPending && !authState.isAuthenticated && <Menu.Item onClick={login}>Login</Menu.Item>}
```
Comment (swiftone) [inline custom-login/src/Navbar.jsx:46]: Is the `!authState.isAuthenticated` redundant and distracting here?

Comment (shuowu-okta) [reply custom-login/src/Navbar.jsx:46]: It's in Navbar which is not protected by SecureRoute. Probably still need to keep the logic here.

**Comment by mraible** [comment]
This seems like a major change. Does okta-react v4 not pull in okta-auth-js?

**Comment by shuowu** [comment]
@mraible okta-react v4 accepts oktaAuth instance as a props from `<Security>` comp instead of creating the instance implicitly.

File: custom-login/src/Navbar.jsx (hunk 4 above)
Comment (swiftone) [reply custom-login/src/Navbar.jsx:46]: But when will isAuthenticated be true if isPending is true?

File: custom-login/src/Login.jsx (hunk 3 above)
Comment (shuowu-okta) [reply custom-login/src/Login.jsx:46]: I saw Aaron has a separated PR for this https://github.com/okta/samples-js-react/pull/146
Let's merge from there.

File: custom-login/src/Navbar.jsx (hunk 4 above)
Comment (shuowu-okta) [reply custom-login/src/Navbar.jsx:46]: I think the logic here is to show `Login` button when not authenticated, so we'll need the `authState.isAuthenticated` flag. Also, we don't want to show the button when in pending state, so we use both.

**Comment by shuowu-okta** [comment]
Screen Shot 2020-11-23 at 12 42 31 PM

**Review by swiftone — APPROVED** [review]

**Pull request closed without merging**